### PR TITLE
group-split

### LINKED ISSUE #21: Split by group membership

### PR DESCRIPTION
**Pull request by ycleptkellan**
closes #21

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,6 +53,8 @@
     "@types/react-select": "^1.0.59",
     "@types/uuid": "^2.0.30",
     "@types/webpack-env": "^1.13.1",
+    "@ycleptkellan/substantive": "^1.0.2",
```
Comment (ericnewcomer) [inline package.json]: What's going on here? Is this a fork?

File: src/components/Flow.test.tsx [hunk 2]
```diff
@@ -18,7 +18,9 @@ const props: FlowProps = {
     nodeDragging: false,
     onDrag: jest.fn(),
     language,
-    translating: baseLanguage.iso !== language.iso && baseLanguage.name === language.name,
+    translating:
+        baseLanguage.iso !== language.iso &&
+        baseLanguage.name === language.name,
     definition,
```
Comment (ericnewcomer) [inline src/components/Flow.test.tsx]: Man, just tons of reformatting like this in this PR. Cool with tweaking stuff you're touching, but if you want to do this much reformatting, it really belongs in it's own PR that only does that.

File: src/components/NodeEditor/NodeEditor.tsx [hunk 3]
```diff
@@ -596,7 +648,10 @@ export default class NodeEditor extends React.PureComponent<NodeEditorProps, Nod
                     key={'fc-back'}
                     onKeyPress={this.onKeyPress}
                     styles={formStyles.advanced}>
-                    <FormComp ref={this.formRef} {...{ ...formProps, showAdvanced: true }} />
+                    <FormComp
+                        ref={this.onFormRef}
+                        {...{ ...formProps, showAdvanced: true }}
```
Comment (ericnewcomer) [inline src/components/NodeEditor/NodeEditor.tsx]: double spread 🙀

File: package.json (hunk 1 above)
Comment (ycleptkellan) [reply package.json]: i moved some helpers to their own library https://github.com/ycleptkellan/substantive

File: src/components/Flow.test.tsx (hunk 2 above)
Comment (ycleptkellan) [reply src/components/Flow.test.tsx]: i think most if not all of the reformatting is introduced by [this commit](https://github.com/nyaruka/floweditor/pull/129/commits/ffc00454b35fd6b7783b88713f9739b42fc8c4bc), happy to revert.

File: src/components/NodeEditor/NodeEditor.tsx (hunk 3 above)
Comment (ycleptkellan) [reply src/components/NodeEditor/NodeEditor.tsx]: hah not a fan? it's an easy way to overwrite props in tests but can understand if it's something you want avoid

**Comment by ycleptkellan** [comment]
Aye! Changes forthcoming
